inst/include/child_model_simulation.hpp

```diff
@@ -91,10 +446,71 @@ void run_child_hiv_infections(int time_step,
           if (cpars.hc1_cd4_dist(hd) > 0) {
             state_next.children.hc1_hiv_pop(hd, 0, a, s) += state_next.base.p_infections(a, s) * cpars.hc1_cd4_dist(hd);
           }
-        }
+        } // end hc1DS
       }
-    }
-  }
+    } //end a
+  } // end NS
+
+  internal::run_calculate_perinatal_transmission_rate(time_step, pars, state_curr, state_next, intermediate);
+  //Perinatal transmission
+  for (int s = 0; s < ss.NS; ++s) {
+  for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
+       state_next.children.hc1_hiv_pop(hd, 0, 0, s) +=  state_next.children.hiv_births * intermediate.children.perinatal_transmission_rate * demog.births_sex_prop(s,time_step) * cpars.hc1_cd4_dist(hd);
+      }// end hc1DS
+      state_next.base.p_hiv_pop(0, s) +=  state_next.children.hiv_births * intermediate.children.perinatal_transmission_rate * demog.births_sex_prop(s,time_step);
+      state_next.base.p_infections(0, s) += state_next.children.hiv_births * intermediate.children.perinatal_transmission_rate * demog.births_sex_prop(s,time_step);
+
+  }// end NS
+
+
+   //Breastfeeding transmission
+
+   //0-6
+   internal::adjust_optAB_bf_transmission_rate(time_step, pars, state_curr, state_next, intermediate);
+    internal::run_calculate_transmission_from_incidence_during_breastfeeding(time_step, pars, state_curr, state_next, intermediate);
+    internal::run_bf_transmission_rate(time_step, pars, intermediate, 0, 3, 0);
+   for (int s = 0; s < ss.NS; ++s) {
+     for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
+       state_next.children.hc1_hiv_pop(hd, 1, 0, s) +=  state_next.children.hiv_births *  demog.births_sex_prop(s,time_step) * cpars.hc1_cd4_dist(hd) * (intermediate.children.bf_incident_hiv_transmission_rate + intermediate.children.bf_transmission_rate(0));
+     }// end hc1DS
+     state_next.base.p_hiv_pop(0, s) +=  state_next.children.hiv_births  * demog.births_sex_prop(s,time_step) * (intermediate.children.bf_incident_hiv_transmission_rate + intermediate.children.bf_transmission_rate(0));
+     state_next.base.p_infections(0, s) += state_next.children.hiv_births  * demog.births_sex_prop(s,time_step) * (intermediate.children.bf_incident_hiv_transmission_rate + intermediate.children.bf_transmission_rate(0));
+   }// end NS
+
+   //6-12
```

Comment:
I think this reads quite nicely. You could split this function up further into separate 0-6, 6-12, 12+ but I'm not sure if that would really make it easier to read. 

The deeply nested names in this context can obfuscate the actual content though a bit, I might have a look at namespacing those at the top of the function and seeing how much that tidies it up. But I can have a look at that in a separate PR later.